### DISCUSSION TITLE
Organization Archiving Beta Feedback

### DISCUSSION THREAD

**Discussion by ahpook** [community/community#58162] [Repositories] [answered]
We recently launched the ability to archive an entire organization - more info here: 

https://docs.github.com/en/organizations/managing-organization-settings/archiving-an-organization

If you've got old/unused organizations that need to go into read-only mode, but the prospect of going through and manually archiving all the repos has been daunting, give this a try and let us know how it works for you!

**Comment by ghost** [comment]
It would be nice if the archive state for the organization showed in any lists of organizations, such as on https://github.com/settings/organizations.

**Reply by lumaxis** [reply]
That makes a lot of sense! We already have this on our backlog, I'll see if we can add it soon.

**Reply by souravcipher** [reply]
+1 

An `Archived` tag next to the organization name would be awesome!
here: https://github.com/settings/organizations

**Reply by helen** [reply]
Labels in lists definitely makes sense. I wonder if for that particular list in settings it would make sense to break archived organizations out into a separate grouping. Any thoughts there?

**Reply by lumaxis** [reply]
@mlewinsk-quest @souravcipher We are now showing labels on that page as you suggested. Please take a look at let us know what you think!

**Comment by Marcono1234** [comment]
As user / consumer of an archived repository (or now a whole organization), I would love if there was a uniform way for the maintainers to explain _why_ the repository or organization was archived (see [my original suggestion](https://github.com/orgs/community/discussions/5383)). Currently maintainers specify the reason in the "About" section of the repository, the README in some non-uniform way, or not at all. And because updating the README requires that the repository is not archived, this new feature of archiving a whole organization might make it even more unlikely that maintainers will update the project's README retroactively to explain the reason for the archiving, because they would have to unarchive the whole organization first, if I understand it correctly.

**Reply by helen** [reply]
Noting that this is an Enterprise Cloud/Enterprise Server feature, does this match what you have in mind or otherwise address the concern? https://github.blog/changelog/2023-01-10-enterprises-and-organizations-can-display-critical-announcements/

Because it's not a globally available feature, I recognize that it may not solve for your problem, this is to help build an understanding of what does or doesn't work :)

**Reply by Marcono1234** [reply]
Thanks for the hint! I am not familiar with GitHub Enterprise, but if I understand it correctly, these are separate GitHub instances, and not the same as github.com, right? In that case I don't think it would directly help me since these announcements would not be visible to all github.com users who look at an organization, but I can imagine it would be helpful for enterprises.
I wrote my previous comment from the perspective of a user who "consumes" the content of an archived repository or who stumbles across an archived repository and would like to know what happened. I did not write it from the perspective of someone who wants to archive a repository (or organization), and was just hoping that if the support for such an "archiving note" existed, users who archive a repository would use it to explain the reason. But this is of course not guaranteed, but the archiving UI could encourage it, see my original suggestion.

Maybe the importance of such an "archiving note" is not actually that great; it just happened a few times in the past that I stumbled upon an archived repository, and mostly out of curiosity would have liked to understand why the repository was archived. Often this was described somewhere, but not in a uniform way.
It might be interesting to hear the feedback of users who did archive an repository in the past; I would imagine especially the fact that the README is currently the most appropriate way to explain the reason for archiving, but updating the README requires unarchiving is quite cumbersome; which is why I suggested this separate "archiving note" feature.

Sorry for hijacking this discussion if my comment was too unrelated to this new feature of archiving organizations.

**Reply by lumaxis** [reply]
>these are separate GitHub instances, and not the same as github.com

@Marcono1234 No, not necessarily! There is an on-premise version of GitHub (called GitHub Enterprise Server) but generally, what we refer to as "GitHub Enterprise" is GitHub Enterprise Cloud which is a specific pricing plan on github.com: https://docs.github.com/en/get-started/onboarding/getting-started-with-github-enterprise-cloud

We definitely did discuss adding something like the "archiving note" feature you mentioned. For the moment, I do think just using the [organization Readme](https://docs.github.com/en/organizations/collaborating-with-groups-in-organizations/customizing-your-organizations-profile) is probably the most straight forward solution. Just as an idea, maybe we could prompt users to update it before archiving?
This is exactly why we're gathering feedback here so thanks for the input!

**Reply by Marcono1234** [reply]
Thanks for the information and the correction!

> Just as an idea, maybe we could prompt users to update it before archiving?

I don't have an organization where I can currently test this, but according to the [documentation](https://docs.github.com/en/repositories/archiving-a-github-repository/archiving-repositories#archiving-a-repository) (see screenshot there), when archiving a repository it suggests to update the README before. Do you mean the same should be suggested when archiving an organization as well (in case it isn't yet)? Or is your suggestion even going one step further to have an edit link or even a Markdown editor to directly update the README during archiving?

**Reply by lumaxis** [reply]
Yes, I think all of these could be viable options to try!

**Reply by ahpook** [reply]
For the time being, we opted to just add some [documentation hints](https://docs.github.com/en/organizations/managing-organization-settings/archiving-an-organization) to this effect:

> You can archive your organization to make it read-only and indicate that it's no longer actively maintained. Prior to archiving, consider updating your organization's profile and README with context about why it's archived. To add an organization-wide README, see "[Customizing your organization's profile](https://docs.github.com/en/organizations/collaborating-with-groups-in-organizations/customizing-your-organizations-profile)."

**Comment by Katsute** [comment]
Feature doesn't seem to work on my organization @Ktt-Development, the option is disabled even though the organization is under the free plan.

![archived organization button grayed out](https://github.com/community/community/assets/58778985/fd4afede-8d66-4679-87a5-64ea8dbf21eb)

![organization free plan](https://github.com/community/community/assets/58778985/d8c52bd3-d4e3-4007-b599-a8088e57bd5c)

**Reply by ahpook** [reply]
Hi @Katsute - This may be a bug - it looks like all of the repositories in your org are already archived. If you unarchive one (or make a new, active repo), does the button become active?

**Reply by Katsute** [reply]
I tried that before but the button didn't change. My active org @KatsuteDev also has this option disabled.

Creating a new org does however have the archive option available.

**Reply by lumaxis** [reply]
@Katsute Could you please check if the button is enabled for you now? I deployed a change yesterday that should correctly assess the billing in some edge cases 🙏🏼

**Reply by Katsute** [reply]
Button works now 👍 ty

**Comment by jvaneikeren** [comment]
It would be great if once an Organization was archived, the Organization no longer showed up under the list of Organizations on your main profile page. I think having it still show up in https://github.com/settings/organizations is fine, but the idea here would be to remove it from the displayed list of Organizations on your base profile page, or at least make it a configurable option.

**Reply by jvaneikeren** [reply]
@lumaxis @ahpook - Just bumping this, thanks!

**Reply by ahpook** [reply]
Fair point, thanks, I've added it to the backlog - and I appreciate the bump, i'd missed the first notification.

**Reply by jvaneikeren** [reply]
Thanks @ahpook!

**Reply by lumaxis** [reply]
@jvaneikeren Question for clarification as I look into this: You can already [configure visibility of your organization memberhips](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-membership-in-organizations/publicizing-or-hiding-organization-membership) so in a sense, it's already configurable 🤔

Maybe just leaving that as is is the better option as it gives users the control on whether they want a specific archived org to still show up?

**Reply by ahpook** [reply]
I did not realize that it's already possible to do this. Is there still a reason to special-case archived orgs, then? I'm inclined to say no.

**Reply by jvaneikeren** [reply]
@lumaxis @ahpook - Thanks for your message. I agree that the ability to determine if an organization is publicly listed on your profile page is an existing capability. However, what I'm requesting here is that the list of organizations that you see when viewing your own profile page (i.e., privately) are limited to those organizations that are not archived. The idea here would be that when viewing your own private profile, you only see a list of organizations that are effectively "active" (i.e., non-archived). I often use the organizations list on my private profile as a way to jump to a specific organization that I'm working in; but as I am an owner of a number of historical/inactive (and now archived) organizations, these archived organizations clutter the list. What I'm trying to avoid doing is having to create a different GitHub user and transfer ownership of the archived organizations to that user just so that the archived organizations won't be included in the list of organizations on my private profile.

**Comment by ahpook** [comment, marked as answer]
I'm closing this post as we're headed to GA with this feature! New discussion for that is up here: https://github.com/orgs/community/discussions/66054
